python/src/aiconfig/editor/client/src/components/aiconfigReducer.ts

```diff
@@ -108,22 +107,32 @@ function reduceConsolidateAIConfig(
   action: MutateAIConfigAction,
   responseConfig: AIConfig
 ): ClientAIConfig {
+  // Make sure prompt structure is properly updated. Client input and metadata takes precedence
+  // since it may have been updated by the user while the request was in flight
+  const consolidatePrompt = (statePrompt: ClientPrompt) => {
+    const responsePrompt = responseConfig.prompts.find(
+      (resPrompt) => resPrompt.name === statePrompt.name
+    );
+    return {
+      ...responsePrompt,
+      ...statePrompt,
```

Comment:
Just to confirm, this means we use the fields from `responsePrompt` first and whatever is left over that hasn't been defined by `responsePrompt` and exists within `statePrompts, that's what they're going to use?

Reply:
Yes